File: 242. Valid Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())
            return false;
        if (s.length() == 0)
            return true;
        int count_s[26], count_t[26];
        for (int i = 0; i < 26; i++){
            count_s[i] = 0;
            count_t[i] = 0;
        }
        for (int i = 0; i < s.length(); i++){
            count_s[s[i] - 97] += 1;
        }
        for (int i = 0; i < t.length(); i++){
            count_t[t[i] - 97] += 1;
        }
        for (int i = 0; i < 26; i++){
            if (count_s[i] != count_t[i])
                return false;
        }
        return true;
    }
};

File: 2024/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        s_count = defaultdict(int)
        for c in s:
            s_count[c] += 1
        for c in t:
            s_count[c] -= 1
            if s_count[c] < 0:
                return False
        return True
